homework/home6.1/src/MessageDispatcher.java

```diff
@@ -0,0 +1,29 @@
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+
+public class MessageDispatcher {
+    private final Map<Class<?>, Map<UUID, Consumer<?>>> subscribersMap = new ConcurrentHashMap<>();
+
+    public <MessageType> UUID subscribe(Class<MessageType> messageType, Consumer<MessageType> callback) {
+        UUID id = UUID.randomUUID();
+        subscribersMap.computeIfAbsent(messageType, k -> new ConcurrentHashMap<>()).put(id, callback);
+        return id;
+    }
+
+    public <MessageType> void publish(MessageType message) {
+        Class<?> messageType = message.getClass();
+        Map<UUID, Consumer<?>> subscribers = subscribersMap.get(messageType);
+        if (subscribers != null) {
+            for (Consumer<?> subscriber : subscribers.values()) {
+                ((Consumer<MessageType>) subscriber).accept(message);
```

Comment:
the same concern as for C++ version - callback should be invoked in async method as usually publish method is expected to be instantly or almost instantly executed